### DISCUSSION TITLE
TextBox SelectionStart > SelectionEnd

### DISCUSSION THREAD

**Discussion by zii-dmg** [AvaloniaUI/Avalonia#12020] [General]
Avalonia 11 rc 2.2

Is it expected that `TextBox.SelectionStart` can be greater than `TextBox.SelectionEnd` when we selected text from right to left?

https://github.com/AvaloniaUI/Avalonia/assets/2846896/0da5b6c9-6083-4202-9098-e16c3d83994a

Found some code that expects this behavior https://github.com/AvaloniaUI/Avalonia/blob/a182e1fd98edf858970a9efaf5dc24ffa950a41b/src/Avalonia.Controls/TextBox.cs#L976
but for users it can be strange.

For example, FluentAvalonia, not expects this https://github.com/amwx/FluentAvalonia/blob/31140cccf95f9af428d90767d9f192ed6c8492f5/src/FluentAvalonia/UI/Controls/CommandBarFlyout/TextCommandBarFlyout.cs#L124
https://github.com/amwx/FluentAvalonia/blob/31140cccf95f9af428d90767d9f192ed6c8492f5/src/FluentAvalonia/UI/Controls/CommandBarFlyout/TextCommandBarFlyout.cs#L127

**Comment by Gillibald** [comment]
We could just switch start and end when start > end

**Reply by rabbitism** [reply]
> We could just switch start and end when start > end

Out of curiosity, is this valid in rtl scenarios?

**Reply by Gillibald** [reply]
Start and End are logical positions not visual. It is valid all the time. So start and end could be at any position visually.

**Reply by StefanKoell** [reply]
I know it's an old thread but it seems to be buggy when you use the keyboard to shift-select text "backwards", the SelectionStart and SelectionEnd are both at the "end" (CaretPosition). If you select the text "forward", it works as expected. (I'm using the latest nightly build though).

**Reply by Gillibald** [reply]
> I know it's an old thread but it seems to be buggy when you use the keyboard to shift-select text "backwards", the SelectionStart and SelectionEnd are both at the "end" (CaretPosition). If you select the text "forward", it works as expected. (I'm using the latest nightly build though).

I can't reproduce this

**Reply by StefanKoell** [reply]
@Gillibald The behavior seems to show up as soon as the textbox is losing it's focus. The selection is then "lost". I came to notice it because I have a custom right-click menu item which shows a dialog allowing you to insert/replace text at the caret position/selection. So it seems to be a different issue as described here. Sorry for the confusion. Should I open a new issue or is there a workaround?

**Reply by Gillibald** [reply]
> @Gillibald The behavior seems to show up as soon as the textbox is losing it's focus. The selection is then "lost". I came to notice it because I have a custom right-click menu item which shows a dialog allowing you to insert/replace text at the caret position/selection. So it seems to be a different issue as described here. Sorry for the confusion. Should I open a new issue or is there a workaround?

Please create a new issue

**Reply by StefanKoell** [reply]
@Gillibald before I do that I want to clarify something. Setting `ClearSelectionOnLostFocus` to `False` keeps the selection when focus is lost. So the question is: is the current behavior actually by design maybe?

**Reply by Gillibald** [reply]
There is no bug if the selection isn't lost when ClearSelectionOnLostFocus is false. So yes this is by design.

**Reply by StefanKoell** [reply]
Thanks for clearing that up. For me the issue is resolved then.
